docs/dev/RELEASE.md

```diff
@@ -0,0 +1,87 @@
+## Introduction
+This document guides maintainers through the process of creating an Envoy Gateway release.
+
+## Prerequisites
+- Permissions to push to the Envoy Gateway repository.
+
+## Creating a Minor Release
+
+1. Clone the repo, checkout the `main` branch, ensure it’s up-to-date, and your local branch is clean.
+2. Create a changelog corresponding to the release number that summarizes the changes included in the release.
+   Reference previous [changelogs][changelogs] for additional details.
+3. Submit a [Pull Request][pr] to merge the changelog into the main branch. This should be the last commit to main
+   before cutting the release.
+4. Create a new release branch from `main`. The release branch should be named
+   `release/v${MAJOR_VERSION}.${MINOR_VERSION}.0`, e.g. `release/v0.3.0`.
```

Comment:
I don't think it should have the `.0`. Once a n `x.y.z` comes out, any necessary patch releases to it will be `x.y.(z+1)`, so they'd still be in the same `x.y` release train.  And it'd be misleading for z≠0 patch releases to be on a `release/vX.Y.0` branch.

Reply:
I agree with what you're saying @danehans, but in that case, I think it makes sense to name the release branches `release/vx.y` rather than `release/vx.y.0`, since `vx.y.0` and any other `vx.y.z` versions will be _tags_ inside the `relese/vx.y` _branch_.